### PR TITLE
Don't allow the same expression in ORDER BY with and without WITH FILL

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Don't allow the same expression in ORDER BY with and without WITH FILL. Such invalid expression could lead to logical error `Invalid number of rows in Chunk`.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 0af1bb2eaa45e08c79e2687fa8d8fe40680b78de with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61667/0af1bb2eaa45e08c79e2687fa8d8fe40680b78de/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Processors/Transforms/FillingTransform.cpp [hunk 1]
```diff
@@ -237,10 +237,21 @@ FillingTransform::FillingTransform(
...
+            ordinary_sort_positions.insert(header_.getPositionByName(desc.column_name));
+    }
+
+    std::unordered_set<size_t> unique_positions;
     for (auto pos : fill_column_positions)
+    {
         if (!unique_positions.insert(pos).second)
             throw Exception(ErrorCodes::INVALID_WITH_FILL_EXPRESSION, "Multiple WITH FILL for identical expressions is not supported in ORDER BY");
+        if (ordinary_sort_positions.contains(pos))
+            throw Exception(ErrorCodes::INVALID_WITH_FILL_EXPRESSION, "ORDER BY containing the same expression with and without WITH FILL is not supported");
```
Comment (devcrafter) [inline src/Processors/Transforms/FillingTransform.cpp:253]:
```suggestion
            throw Exception(ErrorCodes::INVALID_WITH_FILL_EXPRESSION, "ORDER BY containing the same expression with and without WITH FILL modifier is not supported");
```

**Review by devcrafter — COMMENTED** [review]
Can we add the query from https://github.com/ClickHouse/ClickHouse/issues/57609#issuecomment-1845176448 as well, or it's different case?

**Comment by Avogar** [comment]
> Can we add the query from https://github.com/ClickHouse/ClickHouse/issues/57609#issuecomment-1845176448 as well, or it's different case?

It's different case and related only to analyzer (I couldn't figure out the reason)

**Review by devcrafter — APPROVED** [review]

**Pull request merged**